diff-pr.sh

```diff
@@ -251,16 +223,36 @@ copy-tar() {
 	done
 }
 
+# a mimic of "bashbrew cat" which should sort slightly more deterministically (so even full-order-changing PRs should have reasonable diffs)
+_bashbrew-cat() {
+	local img first=1
+	for img; do
+		if [ -n "$first" ]; then
+			first=
+		else
+			echo; echo
+		fi
+		bashbrew cat --format '{{ printf "%s\n" (.Manifest.Global.ClearDefaults defaults) }}' "$img"
+		bashbrew list --uniq "$img" \
+			| sort -V \
+			| xargs -r bashbrew list --uniq --build-order \
+			| xargs -r bashbrew cat --format '
+				{{- range $e := .TagEntries -}}
+					{{- printf "\n%s\n" ($e.ClearDefaults $.Manifest.Global) -}}
+				{{- end -}}
+			'
+	done
+}
```

Comment:
This function is made possible by https://github.com/docker-library/bashbrew/commit/0fc0b0968a8bdac3f461a5b889b671f5d72fa154 which allows me to reimplement https://github.com/docker-library/go-dockerlibrary/blob/c5db18cf68eb1bd2c52046d244bb629d3ed3b109/manifest/rfc2822.go#L229-L242, but with `sort -V` piped to `--build-order` in the middle which should give us ~version sorted entries while still being in the right "build order" and still looking just like `bashbrew cat foo` (but such that a complete reordering of `library/foo`'s entries should give ~the same output :smile: :metal:).

Reply:
Hmm, if I pipe the output of this through something like `sed 's/, /,\n  /g'`, we get diffs like this:

```diff
diff --git a/_bashbrew-cat b/_bashbrew-cat
index 22f7cfc..27a895e 100644
--- a/_bashbrew-cat
+++ b/_bashbrew-cat
@@ -1,48 +1,42 @@
 Maintainers: Tianon Gravi <admwiggin@gmail.com> (@tianon), Joseph Ferguson <yosifkit@gmail.com> (@yosifkit)
 GitRepo: https://github.com/docker-library/gcc.git
 
-Tags: 6.5.0,
-  6.5,
-  6
-Architectures: amd64,
-  arm32v5,
-  arm32v7
-GitCommit: e17fd3097b743216f292e50ea8e84b3b3bcc4e53
-Directory: 6
-
-Tags: 7.5.0,
-  7.5,
-  7
+Tags: 8.4.0,
+  8.4,
+  8
 Architectures: amd64,
   arm32v5,
   arm32v7,
   arm64v8,
+  mips64le,
   ppc64le,
   s390x
-GitCommit: aab71caeb1be048b44fd140a102cb7f17f995276
-Directory: 7
+GitCommit: 05aef2fc627328e12bbf77aca44fd399a22c7fc4
+Directory: 8
 
-Tags: 8.4.0,
-  8.4,
-  8
+Tags: 9.3.0,
+  9.3,
+  9
 Architectures: amd64,
   arm32v5,
   arm32v7,
   arm64v8,
+  mips64le,
   ppc64le,
   s390x
-GitCommit: f3db0f591ce42dd2ac3aec240ca47c87762bc385
-Directory: 8
+GitCommit: 05aef2fc627328e12bbf77aca44fd399a22c7fc4
+Directory: 9
 
-Tags: 9.3.0,
-  9.3,
-  9,
+Tags: 10.1.0,
+  10.1,
+  10,
   latest
 Architectures: amd64,
   arm32v5,
   arm32v7,
   arm64v8,
+  mips64le,
   ppc64le,
   s390x
-GitCommit: d9a58942c8ff2c5c5a01e16bf0ffdf5e5dc61277
-Directory: 9
+GitCommit: 97b046b578bd86cae5414d80b3ad0027c590aebd
+Directory: 10
```

I'm not convinced it's better, but it does have some nice properties.

Reply:
That does seem interesting, but I'd vote to stick with the unedited output.

Reply:
Would it make sense to split this out to a separate script in case someone wanted to get this "sorted" bashbrew cat output?

Reply:
For some reason I completely missed this second comment in my email. :confused:

Yeah, I think that's not a bad idea, although ideally it would live in the https://github.com/docker-library/bashbrew repo, which would complicate using it here.  I guess we'll start with it being a script here and go from there.